java/ql/lib/semmle/code/java/frameworks/apache/Codec.qll

```diff
@@ -0,0 +1,135 @@
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,File);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,InputStream);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,Path,OpenOption[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,RandomAccessFile);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,byte[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;HmacUtils;false;updateHmac;(Mac,InputStream);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;HmacUtils;false;updateHmac;(Mac,String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;HmacUtils;false;updateHmac;(Mac,byte[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;Md5Crypt;true;md5Crypt;(byte[],String,String);;Argument[2];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;Md5Crypt;true;md5Crypt;(byte[],String,String,Random);;Argument[2];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Lang;true;loadFromResource;(String,Languages);;Argument[1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Languages$LanguageSet;true;from;(Set);;Element of Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Languages$LanguageSet;true;merge;(LanguageSet);;Argument[-1];ReturnValue;value;",
+        "org.apache.commons.codec.language.bm;Languages$LanguageSet;true;merge;(LanguageSet);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Languages$LanguageSet;true;restrictTo;(LanguageSet);;Argument[-1];ReturnValue;value;",
+        "org.apache.commons.codec.language.bm;Languages$LanguageSet;true;restrictTo;(LanguageSet);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Languages$SomeLanguages;false;getLanguages;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Languages;true;getLanguages;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;PhoneticEngine;true;getLang;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Rule$Phoneme;false;Phoneme;(CharSequence,LanguageSet);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language.bm;Rule$Phoneme;false;Phoneme;(CharSequence,LanguageSet);;Argument[1];Argument[-1];taint;",
+        "org.apache.commons.codec.language.bm;Rule$Phoneme;false;Phoneme;(Phoneme,Phoneme,LanguageSet);;Argument[2];Argument[-1];taint;",
```

Comment:
If a `Phoneme` can be tainted, we should consider arguments 0 and 1 of this constructor and the `Phoneme(Phoneme,Phoneme)` one.